kartothek/io/dask/dataframe.py

```diff
@@ -390,15 +395,9 @@ def collect_dataset_metadata(
                 for mp in mps
             ]
         )
-        df = df.compute()
-        df.sort_values(
-            by=["partition_label", "row_group_id"], ignore_index=True, inplace=True
-        )
     else:
-        warnings.warn(
-            f"Can't retrieve metadata for empty dataset with uuid `{dataset_uuid}`"
-        )
-        df = pd.DataFrame(columns=MetaPartition._METADATA_COLUMNS)
-        df = df.astype(MetaPartition._METADATA_SCHEMA._asdict())
+        df = pd.DataFrame(columns=_METADATA_SCHEMA.keys())
+        df = df.astype(_METADATA_SCHEMA)
+        df = dask.delayed(df)
```

Comment:
This should also be a `dask.DataFrame`, not a delayed object, see https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.from_pandas

```suggestion
        df = dd.from_pandas(
            data=df,
            npartitions=1,
        )
```

Reply:
Why was this working then?

Reply:
Calling compute on delayed also works, of course. And the result returns a pandas dataframe. so this is _almost_ the same

Reply:
It is. See last commit